modules/feature-upload.php

```diff
@@ -291,7 +304,7 @@
 				exit();
 			}
 			
-			$filepath = $env->storage_prefix . $pageindex->{$env->page}->uploadedfilepath;
+			$filepath = getcwd() . "/" . $env->storage_prefix . $pageindex->{$env->page}->uploadedfilepath;
```

Comment:
Yeah, imagick appears to take relative paths just fine IIRC.

And reading this again with fresh eyes, the `storage_directory` is meant to be a path specified by the user that should point to the directory in which the wiki's data is actually stored. e.g. if `index.php` is stored in `/srv/www/sekretapplewiki`, but the data is actually stored in `/srv/data/applestuff`, then `storage_prefix` could either be `/srv/data/applestuff` or `../../data/applestuff`. IIRC it defaults to `.`.

This does rely on the current directory being the directory in which `index.php` is stored, which I'm not sure how often that is the case (it has been for me).

Thanks for your effort! It's been quite enjoyable seeing someone else who's as interested in this project as I am :smiley_cat: